### ISSUE TITLE
[LINUX] Unable to queue up for a match while using Steam beta client

### ISSUE THREAD

**Issue by rmbleeker** [ValveSoftware/Dota-2#1477]
#### Your system information

* System information: [Ubuntu 16.04.4 LTS on a Core i5 8400 with 16GB ram and an NVidia GTX1070](https://gist.github.com/rmbleeker/bb03f96e8a904ba6e756b70956595964)
* Have you checked for system updates?: Yes
* Are you using the latest stable video driver available for your system? Yes, 396.24.02 from https://launchpad.net/~graphics-drivers/+archive/ubuntu/ppa
* Have you verified the game files?: Yes

#### Please describe your issue in as much detail as possible:
While enrolled in the Steam Client Beta update I am unable to search for a Dota 2 match. I can click the "Play Dota" button and select a game type from the sidebar, but after pressing the green "Find Match" button it just turns into an unclickable gray "Play Dota" button instead of counting up and showing the red cancel button while finding a match. The same behavior occurs whether I'm trying to find a solo match or with others. 

If someone else in the party initiates matchmaking it works fine. 

Dota 2 matchmaking works fine again after opting out of the Steam Client Beta.

#### Steps for reproducing this issue:

1. Opt in to the Steam Beta Update (Steam > Settings > Account > Change > Steam Beta Update)
2. Restart Steam as requested in the dialog window
3. Start Dota 2 from the game library
4. Click the "Play Dota" button in the lower right corner
5. Select any game type, for example Unranked > All Pick
6. Click the green "Fund Match" button in the lower right corner
7. Nothing happens, "Play Dota" will remain grayed out until you restart Dota 2

#### Steps to successfully find a match

1. Opt out of the Steam Beta Update (Steam > Settings > Account > Change > NONE - Opt out of all beta programs)
2. Restart Steam as requested in the dialog window
3. Start Dota 2 from the game library
4. Click the "Play Dota" button in the lower right corner
5. Select any game type, for example Unranked > All Pick
6. Click the green "Fund Match" button in the lower right corner
7. Wait while Dota 2 finds a match or cancel matchmaking by clicking the red button in the lower right corner

**Comment by zpostfacto** [comment]
We've been trying to reproduce this problem at the office but have been unable for some reason.

Can you share the output of the Dota console?  Also, if you launch Steam from a terminal (~/.local/share/Steam/steam.sh), can you share the output of that terminal session?

Also, you should have a file ~/.local/share/Steam/network_config.json.  Can you share that file?

**Comment by rmbleeker** [comment]
The terminal output from Steam should be logged to the file /tmp/dumps/<username>_stdout.txt if I'm not mistaken, so how about I just share that file?

I'll start Steam twice, once subscribed to the beta and once while I'm not and give you the outputs of both sessions.

**Comment by kisak-valve** [comment]
Hello @rmbleeker, you can put the log in a [gist](https://gist.github.com/) or drag and drop it directly into the comment box since it is a txt file.

**Comment by zpostfacto** [comment]
@rmbleeker, sure thing.

**Comment by rmbleeker** [comment]
[output.zip](https://github.com/ValveSoftware/Dota-2/files/2161239/output.zip)
Here's a zip file containing the Steam console output and Dota 2 console output from two sessions, one in the beta client and one in the stable client. There is also a crash log that I happened to get while switching from the Steam beta client to the stable client. That was a fluke though, since I've switched between them many times today and yesterday and this is the first time it happened. I hope it doesn't mess with tracking down the cause of this.

**Comment by zpostfacto** [comment]
We think it might related to Ubuntu 16.  We are on 18 here, that's probably why we cannot reproduce it.

**Comment by zpostfacto** [comment]
@rmbleeker, are you able to wget this URL?

https://steamcdn-a.akamaihd.net/apps/sdr/network_config.json

**Comment by rmbleeker** [comment]
$ wget https://steamcdn-a.akamaihd.net/apps/sdr/network_config.json
--2018-07-04 01:02:06--  https://steamcdn-a.akamaihd.net/apps/sdr/network_config.json
Resolving steamcdn-a.akamaihd.net (steamcdn-a.akamaihd.net)... 104.93.82.232, 104.93.82.186
Connecting to steamcdn-a.akamaihd.net (steamcdn-a.akamaihd.net)|104.93.82.232|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 16208 (16K) [application/json]
Saving to: ‘network_config.json.1’

network_config.json.1             100%[=============================================================>]  15,83K  --.-KB/s    in 0s      

2018-07-04 01:02:06 (149 MB/s) - ‘network_config.json.1’ saved [16208/16208]

**Comment by rmbleeker** [comment]
Well it's as good a reason as any to upgrade to 18.04 then I guess.

**Comment by zpostfacto** [comment]
@rmbleeker, do you know what locale you are using?  E.g. try:

env | egrep -i '(lc|lang)

**Comment by zpostfacto** [comment]
We think it's a LOCALE ISSUE.  There is a field (strictly for human consumption...lol) where we describe each point-of-presence, and one of them is in......São Paulo.  And that is why we couldn't reproduce it, because we all have our locale set to US default or whatever.

I just updated the config.  See if it works now!

**Comment by xfalcox** [comment]
Hi @fletcherdvalve 

I'm using Ubuntu 18.04 and having the same issue from São Paulo too.

Anything I can do to help debug the issue?

**Comment by zpostfacto** [comment]
@xfalcox LoL, thanks but it doesn't actually matter where you are from.

And it looks like it might be related to just parsing and printing numbers and doesn't have anything to do with the extended characters.  Something in the pipe is being locale-specific in a dumb way.

**Comment by xfalcox** [comment]
My locale related env variables are:

```
HOME=/home/falco
LANG=en_US.UTF-8
LC_ADDRESS=pt_BR.UTF-8
LC_CTYPE=pt_BR.UTF-8
LC_IDENTIFICATION=pt_BR.UTF-8
LC_MEASUREMENT=pt_BR.UTF-8
LC_MONETARY=pt_BR.UTF-8
LC_NAME=pt_BR.UTF-8
LC_NUMERIC=pt_BR.UTF-8
LC_PAPER=pt_BR.UTF-8
LC_TELEPHONE=pt_BR.UTF-8
LC_TIME=pt_BR.UTF-8
```

Also, it may help to debug that the server selection modal also doesn't work (it will keep on "Calculating..." forever).

**Comment by xfalcox** [comment]
Also here is my output from steam @fletcherdvalve 

https://gist.github.com/xfalcox/aa734056767e931feb24e76a2dc255af

**Comment by zpostfacto** [comment]
Can everybody having this bug confirm that setting LC_ALL=C fixes the problem?

We think the correct solution is just to totally disable all the CRT locale code, which we have done on other platforms but for some reason not Linux.  Want to make sure it does actually fix the problem and not break anything worse.

**Comment by zpostfacto** [comment]
We are planning on updating the beta with this change sometime int he next few days.

**Comment by xfalcox** [comment]
> Can everybody having this bug confirm that setting LC_ALL=C fixes the problem?

Yes, it fixes it! :tada: 

I can use the server ping modal and search for a match now.

**Comment by lordtiva** [comment]
Not working for me, using Arch Linux + KDE

**Comment by xfalcox** [comment]
@lordtiva you want to launch steam as:

```
LC_ALL=C ~/.local/share/Steam/steam.sh
```

**Comment by rmbleeker** [comment]
@fletcherdvalve I see you've made progress while I was asleep, but just in case it still matters somehow I've pasted the output of the "locale" command below. As you can see I'm using a mixed locale, en_US.UTF-8 as a base with some regional settings (The Netherlands) for some number representation and stuff like that.

$ locale
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=nl_NL.UTF-8
LC_TIME=nl_NL.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=nl_NL.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=nl_NL.UTF-8
LC_NAME=nl_NL.UTF-8
LC_ADDRESS=nl_NL.UTF-8
LC_TELEPHONE=nl_NL.UTF-8
LC_MEASUREMENT=nl_NL.UTF-8
LC_IDENTIFICATION=nl_NL.UTF-8
LC_ALL=

**Comment by rmbleeker** [comment]
It seems that this is indeed a locale related issue. Starting Steam as `LC_ALL=C steam` while enrolled in the client beta enables me to find a match in Dota 2, without that setting it doesn't work. For now I've added `export LC_ALL=C` somewhere at the top of `/usr/bin/steam` as a work-around.

```
#!/usr/bin/env bash
#
# This is the Steam script that typically resides in /usr/bin
# It will create the Steam bootstrap if necessary and then launch steam.

# To work around any fancy locale settings, set it to C
export LC_ALL=C

# Get the full name of this script
...
```

**Comment by SergioGlorias** [comment]
@fletcherdvalve eu sou de portugal e tambem estou com o mesmo problema

**Comment by SergioGlorias** [comment]
e com esse método tambem

**Comment by ghost** [comment]
@rmbleeker thanks you so much. It works like a charm

**Comment by kisak-valve** [comment]
Hello, per "Fixed an issue where CS:GO and Dota 2 matchmaking would not function properly." in the [2018-07-11](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/2642934601290756923) Steam client beta update, please update your steam client and retest this issue.

**Comment by rmbleeker** [comment]
I've just tested this after applying the [2018-07-11](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/2642934601290756923) update to the Steam beta client and can confirm matchmaking in Dota 2 works as expected again. 

I've tested this by removing `export LC_ALL=C` from `/usr/bin/steam`, then I started Steam and Dota 2. I looked up the PID for the Dota process and checked `/proc/<PID>/environ` to see if the locale in there was actually what I was expecting, making sure both _en_US_ and _nl_NL_ were present for their respective locale variables. That was the case, but I was still able to find a Dota 2 match.

Thanks for the quick response and fix, I'm closing this issue now.
